### PR TITLE
refactor: remove circular dependencies through net_processing (1/N)

### PR DESCRIPTION
**Pull request by knst**
## Issue being fixed or feature implemented
The architecture of bitcoin assumes that there's no any external class that processes network messages and knows anything about PeerManager from net_processing; no any external call for `PeerManager::Misbehaving` in bitcoin. All logic related to processing messages are located in net_processing.
Dash has many many extra types of network messages and many of them processed by external components such as llmq/signing or coinjoin/client. Current architecture creates multiple circular dependency.

## What was done?
This PR introduces several key changes:
 - **new helper `tl::expected`** that can be widely used in dash code and it has very similar implementation with `std::expected` and can be replaced to `std::expected` since C++23 is used by default. Please notice that this PR will superseed https://github.com/dashpay/dash/pull/5109
 - new helper `PeerMsgRet` that is defined as `tl::expected<void, MisbehavingError>` inside net_types. That lets external modules that process network messages to don't depend directly on net_processing.
 - removed dependency of many classes on net_processing thanks to new `PeerMsgRet` type: `coinjoin/client`, `coinjoin/server`, `evo/mnauth`, `governance/governance`, `llmq/blockprocessor`, `llmq/chainlocks`, `llmq/quorums`, `spork` 

## What else to do?
Some network messages are processed asynchronously in external components such as llmq/signing, llmq/instantsend, llmq/dkgsessionhandler. It doesn't let to refactor them easily, because they can't just simple return status of processing; status of processing would be available sometime later and there's need callback or other way to pass result code without spreading PeerManager over codebase.

## How Has This Been Tested?
 - run unit functional tests
 - run a linter `test/lint/lint-circular-dependencies.sh`

## Breaking Changes
N/A

## Checklist:
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Comment by PastaPastaPasta** [comment]
Otherwise LGTM

File: src/net_types.h [hunk 1]
```diff
@@ -5,11 +5,28 @@
...
 class CSubNet;
 
 using banmap_t = std::map<CSubNet, CBanEntry>;
 
+struct MisbehavingError
+{
+    int score;
+    std::string message;
+
+    MisbehavingError(int score, std::string message = "") :
```
[the remark this answers is not in the document]
Comment (knst) [reply src/net_types.h:23]: Most of places doesn't have a text message.
Also for case of text message it's little too verbose: need to write `tl::expected<MisbehavingError>(...)`

For other usages (not net_processing) I'd agree if error usually provides text description

File: src/net_types.h [hunk 2]
```diff
@@ -5,11 +5,28 @@
...
+        score{score},
+        message{std::move(message)}
```
[the remark this answers is not in the document]
Comment (knst) [reply src/net_types.h]: updated, please check

**Review by PastaPastaPasta — COMMENTED** [review]
utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK

**Review by PastaPastaPasta — DISMISSED** [review]
re-utACK; please state preference for merge commit vs squash

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

**Comment by knst** [comment]
resolved conflicts and added expections for linter.

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**